lib/api_auth/base.rb

```diff
@@ -96,12 +99,12 @@ def hmac_signature(headers, secret_key, options)
                          else
                            headers.canonical_string
                          end
-      digest = OpenSSL::Digest.new('sha1')
+      digest = OpenSSL::Digest.new(options[:digest])
       b64_encode(OpenSSL::HMAC.digest(digest, secret_key, canonical_string))
     end
 
     def auth_header(headers, access_id, secret_key, options)
-      "APIAuth #{access_id}:#{hmac_signature(headers, secret_key, options)}"
+      "APIAuth-HMAC-#{options[:digest].upcase} #{access_id}:#{hmac_signature(headers, secret_key, options)}"
```

Comment:
I think for now, we have to only add the -HMAC-digest if the digest isn't SHA1, otherwise there is no upgrade path for the users. Users of this gem need to be able to update client and server independently. And it should be backwards compatible it both directions just in case the "server" ever signs messages that go back to the "client" in the case of web-hooks or some other "server" initiated communication.